sdk/servicebus/azure-messaging-servicebus/migration-guide.md

```diff
@@ -0,0 +1,377 @@
+
+```java
+// Create a sender client that will authenticate through Active Directory
+TokenCredential credential = new DefaultAzureCredentialBuilder()
+    .build();
+String fullyQualifiedNamespace = "yournamespace.servicebus.windows.net";
+ServiceBusSenderClient client = new ServiceBusClientBuilder()
+    .credential(fullyQualifiedNamespace, credential)
+    .sender()
+    .queueName("my-queue")
+    .buildClient();
+
+// Create a sender client that will authenticate using a connection string
+String connectionString = "Endpoint=sb://yournamespace.servicebus.windows.net/;SharedAccessKeyName=your-key-name;SharedAccessKey=your-key";
+ServiceBusSenderClient client = new ServiceBusClientBuilder()
+    .connectionString(connectionString)
+    .sender()
+    .queueName("my-queue")
+    .buildClient();
+```
+
+### Sending messages
+
+Previously, in `azure-servicebus`, you could send messages either by using a `IQueueClient` (or `ITopicClient` if you
+are targeting a topic) or the `IMessageSender`.
+
+While the `IQueueClient` supported the simple send operation, the `IMessageSender` supported that and advanced scenarios
+like scheduling to send messages at a later time and cancelling such scheduled messages.
+
+```java
+String queueName = "my-queue";
+String connectionString = "Endpoint=sb://yournamespace.servicebus.windows.net/;"
+    + "SharedAccessKeyName=your-key-name;SharedAccessKey=your-key";
+
+// create a message to send
+Message message = new Message("content");
+
+// send using the QueueClient
+QueueClient client = new QueueClient(new ConnectionStringBuilder(connectionString, queueName),
+    ReceiveMode.PEEKLOCK);
+client.send(message);
+
+// send using the IMessageSender
+IMessageSender sender = ClientFactory.createMessageSenderFromConnectionStringBuilder(
+    new ConnectionStringBuilder(connectionString, queueName));
+sender.send(message);
+```
+
+Now in `azure-messaging-servicebus`, we combine all send related features under a common class `ServiceBusSenderClient`
+and its async counterpart `ServiceBusSenderAsyncClient`. You can create these from the top-level client builder using the `sender()` method to
+get a sub-builder. The sub builder takes the queue or topic you want to target. This way, we give you a one stop shop for
+all your send related needs.
+
+We continue to support sending bytes in the message. Though, if you are working with strings, you can now create a
+message directly without having to convert it to bytes first. The snippet below demonstrates the sync sender client.
+
+```java
+// create the sync sender via the builder and its sub-builder
+ServiceBusSenderClient client = new ServiceBusClientBuilder()
+    .connectionString(connectionString)
+    .sender()
+    .queueName("my-queue")
+    .buildClient();
+
+// create a message to send
+ServiceBusMessage message = new ServiceBusMessage("Hello world!");
+
+// send the message
+sender.SendMessage(message);
```

Comment:
We'll probably have to use [embedme tool](https://github.com/Azure/azure-sdk-for-java/wiki/Building#code-snippets-in-readme) for injecting these code snippets into the migration guide to ensure they are compilable. 

```suggestion
sender.sendMessage(message);
```